src/SFA.DAS.Data.Database/Views/DAS_LevyDeclarations.sql

```diff
@@ -29,6 +29,11 @@ SELECT LD.[Id]
 	-- Flag to say if latest record from subquery, Using Coalesce to set null value to 0
 
 	, COALESCE(LLD.Flag_Latest,0) AS Flag_Latest 
+     , DATENAME(month,DATEFROMPARTS(CASE WHEN LD.[PayrollMonth] Between 10 and 12 then '20'+ CAST(RIGHT(LD.[PayrollYear],2) AS VARCHAR(255))
+								    ELSE '20'+CAST(LEFT(LD.[PayrollYear],2) AS VARCHAR(255)) END  ,CASE WHEN LD.[PayrollMonth] Between 10 and 12 THEN LD.[PayrollMonth]- 9
+											ELSE LD.[PayrollMonth] +3 END ,1)) +' - ' + CASE WHEN LD.[PayrollMonth] Between 10 and 12 then '20'+ CAST(RIGHT(LD.[PayrollYear],2) AS VARCHAR(255))
```

Comment:
My T-SQL is rusty but can this not be encapsulated in a function? Then rather than burning CPU on computing this how about joining to a lookup table which maps the calendar month+year through to the alternative month+year then just precompute the table for the next 25 years? That's trivial to visual see is correct and you can keep the nasty date math for the spreadsheet you used to generate the lookup table.